src/main/scala/scorex/block/Block.scala

```diff
@@ -90,32 +90,42 @@ case class Block(timestamp: Long, version: Byte, reference: ByteStr, signerData:
 
 
 object Block extends ScorexLogging {
+
+  val PrevBlockFee: Float = 0.6f
+  val CurrentBlockFee: Float = 0.4f
+
   type BlockIds = Seq[ByteStr]
   type BlockId = ByteStr
 
-  val MaxTransactionsPerBlockVer1: Int = 100
-  val MaxTransactionsPerBlockVer2: Int = 65535
+  val MaxTransactionsPerBlockVer1Ver2: Int = 100
+  val MaxTransactionsPerBlockVer3: Int = 65535
   val BaseTargetLength: Int = 8
   val GeneratorSignatureLength: Int = 32
 
   val BlockIdLength = SignatureLength
 
   val TransactionSizeLength = 4
 
-  def transParseBytes(version: Int,bytes: Array[Byte]): Try[Seq[Transaction]] = Try {
-    if (bytes.isEmpty ) {
-       Seq.empty
-      } else {
-        val v: (Array[Byte], Int) = version match {
-        case 1 | 2 => (bytes.tail        , bytes.head) //  255  max
+  def transParseBytes(version: Int, bytes: Array[Byte]): Try[Seq[Transaction]] = Try {
+    if (bytes.isEmpty) {
+      Seq.empty
+    } else {
+      val v: (Array[Byte], Int) = version match {
+        case 1 | 2 => (bytes.tail, bytes.head) //  127 max, won't work properly wif greater
+        case 3 =>
+          // https://stackoverflow.com/a/18247942/288091
+          val high = if (bytes(1) >= 0) bytes(1) else 256 + bytes(1)
+          val low = if (bytes(0) >= 0) bytes(0) else 256 + bytes(0)
+          val amount = low | (high << 8)
```

Comment:
Why little endian? We already use `com.google.common.primitives.Ints/Shorts/...` (e.g. see `scorex.serialization`) which works with big endian.